akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/MultiNodeClusterShardingSpecSpec.scala

```diff
@@ -0,0 +1,125 @@
+
+object DDataClusterShardingRememberEntitiesEnabledConfig
+    extends MultiNodeClusterShardingSpecConfig(ClusterShardingSettings.StateStoreModeDData, true)
+
+class DDataClusterShardingMultiNodeSpecMultiJvmNode1
+    extends MultiNodeClusterShardingSpecSpec(DDataClusterShardingRememberEntitiesEnabledConfig)
+class DDataClusterShardingMultiNodeSpecMultiJvmNode2
+    extends MultiNodeClusterShardingSpecSpec(DDataClusterShardingRememberEntitiesEnabledConfig)
+class DDataClusterShardingMultiNodeSpecMultiJvmNode3
+    extends MultiNodeClusterShardingSpecSpec(DDataClusterShardingRememberEntitiesEnabledConfig)
+class DDataClusterShardingMultiNodeSpecMultiJvmNode4
+    extends MultiNodeClusterShardingSpecSpec(DDataClusterShardingRememberEntitiesEnabledConfig)
+
+abstract class MultiNodeClusterShardingSpecSpec(multiNodeConfig: MultiNodeClusterShardingSpecConfig)
+    extends MultiNodeClusterShardingSpec(multiNodeConfig) {
+  import MultiNodeClusterShardingSpec._, multiNodeConfig._
+
+  val dataType = "Entity"
+
+  def startSharding(sys: ActorSystem, probe: ActorRef): ActorRef =
+    super.startSharding(sys, MultiNodeClusterShardingSpec.props(probe), dataType)
+
+  def expectEntityRestarted(sys: ActorSystem, event: Int, probe: TestProbe, entityProbe: TestProbe): EntityStarted = {
+    if (!rememberEntities) {
+      probe.send(ClusterSharding(sys).shardRegion(dataType), event)
+      probe.expectMsg(1)
+    }
+
+    entityProbe.expectMsgType[EntityStarted](30.seconds)
+  }
+
+  s"Cluster sharding [rememberEntities=$rememberEntities, mode=$mode]" must {
+
+    "restart and rebalance entities on coordinator node down" in within(30.seconds) {
+      startPersistenceIfNotDdataMode(startOn = first, setStoreOn = Seq(first, second, third))
+
+      val entityProbe = TestProbe()
+      val probe = TestProbe()
+      join(second, second)
+      runOn(second) {
+        startSharding(system, entityProbe.ref)
+        probe.send(ClusterSharding(system).shardRegion(dataType), 1)
+        probe.expectMsg(1)
+        entityProbe.expectMsgType[EntityStarted]
+      }
+      enterBarrier("second-started")
+
+      join(third, second)
+      runOn(third) {
+        startSharding(system, entityProbe.ref)
+      }
+
+      join(fourth, second)
+      runOn(fourth) {
+        startSharding(system, entityProbe.ref)
+      }
+
+      runOn(second, third, fourth) {
+        within(remaining) {
+          awaitAssert {
+            cluster.state.members.size shouldEqual 3
+            cluster.state.members.forall(_.status == MemberStatus.Up) shouldEqual true
+          }
+        }
+      }
+      enterBarrier("all-up")
+
+      runOn(first) {
+        if (isDdataMode) {
+          // Entity 1 in region of first node was started when there was only one node
+          // and then the remembering state will be replicated to second node by the
+          // gossip. So we must give that a chance to replicate before shutting down second.
+          Thread.sleep(5000)
```

Comment:
I've tried tuning several settings to not need this. Nothing successfully yet. It's used by another sharding test.

Reply:
I think you can't get rid of it completely but you can tune `akka.cluster.sharding.distributed-data.gossip-interval` down from the default `2s` to make gossip faster, and then perhaps base this on the value of that setting rather than arbitrary magic number of millis, like a 2x or something.

Reply:
Will try elsewhere/later.